app/uk/gov/hmrc/tai/service/TaxAccountSummaryService.scala

```diff
@@ -45,9 +45,11 @@ class TaxAccountSummaryService @Inject() (
     for {
       totalEstimatedTax       <- taxAccountHelper.totalEstimatedTax(nino, year)
       taxFreeAmountComponents <- codingComponentService.codingComponents(nino, year)
-      taxCodeIncomes          <- incomeService.taxCodeIncomes(nino, year)
-      totalTax                <- totalTaxService.totalTax(nino, year)
-      taxFreeAllowance        <- totalTaxService.taxFreeAllowance(nino, year)
+      taxCodeIncomes <- incomeService.taxCodeIncomes(nino, year).recoverWith { case ex: MissingEmploymentException =>
+                          Future.failed(ex)
```

Comment:
Is this necessary? There'll already be a future failed returned by taxCodeIncomes

Reply:
No Actually its not needed. I did "bottoms-up" approach and didn't make the change in the controller before this so had put this. Removed it now